### PR TITLE
Cisco Umbrella indicators context outputs - Support for version 5.0

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17949

## Description
Updated Cisco-umbrella Indicators context outputs to support version 5.0.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/Cisco-umbrella/Cisco-umbrella.py [hunk 1]
```diff
@@ -739,19 +739,40 @@ def get_domain_command():
     # whois data
     whois = []  # type: ignore
     whois = get_whois_for_domain(domain)
+
+    # 5.0 update - context indicators
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/Cisco-umbrella.py]: remove the comment

File: Integrations/Cisco-umbrella/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# [Unreleased]
+
+
+## [19.8.2] - 2019-08-26
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/CHANGELOG.md]: remove this, it should be under unreleased

File: Integrations/Cisco-umbrella/Cisco-umbrella.py [hunk 3]
```diff
@@ -739,19 +739,40 @@ def get_domain_command():
     # whois data
     whois = []  # type: ignore
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/Cisco-umbrella.py]: remove this

File: Integrations/Cisco-umbrella/Cisco-umbrella.py [hunk 4]
```diff
@@ -739,19 +739,40 @@ def get_domain_command():
...
     name_servers = whois.get('nameServers')  # type: ignore
     emails = whois.get('emails')  # type: ignore
+    registrar = {'Name': whois.get('registrarName', '')}  # type: ignore
+    creation_date = first_queried
+    domain_status = whois.get('status', [])  # type: ignore
+    updated_date = whois.get('updated', '')  # type: ignore
+    expiration_date = whois.get('expires', '')  # type: ignore
+
     whois = {
         'Name': whois['domainName'],  # type: ignore
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/Cisco-umbrella.py]: change all these to use `.get()`

File: Integrations/Cisco-umbrella/Cisco-umbrella.py [hunk 5]
```diff
@@ -739,19 +739,40 @@ def get_domain_command():
     # whois data
     whois = []  # type: ignore
     whois = get_whois_for_domain(domain)
+
+    # 5.0 update - context indicators
+    admin = {
+        'Country': whois.get('administrativeContactCountry', ''),  # type: ignore
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/Cisco-umbrella.py]: remove all type ignores

**Comment by kirbles19** [comment]
@guyfreund @Itay4 Need a description of what was done and what to review. This PR is too vague to start on.

File: Integrations/Cisco-umbrella/Cisco-umbrella.py [hunk 6]
```diff
@@ -1371,6 +1399,38 @@ def get_whois_for_domain_command():
             'LastObserved': res['auditUpdatedDate'],
         }
 
+        # 5.0 update - context indicators
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/Cisco-umbrella.py]: remove this

File: Integrations/Cisco-umbrella/Cisco-umbrella.py [hunk 7]
```diff
@@ -1371,6 +1399,38 @@ def get_whois_for_domain_command():
...
+        }
+        registrant = {
+            'Country': res.get('registrantCountry', ''),
+            'Email': res.get('registrantEmail', ''),
+            'Name': res.get('registrantName', ''),
+            'Phone': res.get('registrantTelephone', ''),
+        }
+        creation_date = res.get('created')  # type: ignore
+        registrar = {'Name': res.get('registrarName', '')}
+        domain_status = res.get('status', [])[0] if res.get('status') else None  # or [-1] ?
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/Cisco-umbrella.py]: ?

File: Integrations/Cisco-umbrella/Cisco-umbrella.py [hunk 8]
```diff
@@ -1371,6 +1399,38 @@ def get_whois_for_domain_command():
...
+            'Name': res.get('administrativeContactName', ''),
+            'Phone': res.get('administrativeContactTelephone', '')
+        }
+        registrant = {
+            'Country': res.get('registrantCountry', ''),
+            'Email': res.get('registrantEmail', ''),
+            'Name': res.get('registrantName', ''),
+            'Phone': res.get('registrantTelephone', ''),
+        }
+        creation_date = res.get('created')  # type: ignore
```
Comment (Itay4) [inline Integrations/Cisco-umbrella/Cisco-umbrella.py]: remove type ignore

File: Integrations/Cisco-umbrella/Cisco-umbrella.py (hunk 7 above)
Comment (guyfreund) [reply Integrations/Cisco-umbrella/Cisco-umbrella.py]: It's not the updated file probably, the line changed to:
domain_status = whois.get('status', [])

**Comment by kirbles19** [comment]
@Itay4 see my comment above, what exactly do we have to review on this PR?

**Comment by guyfreund** [comment]
Hey @kirbles19, there are the changes needed to be reviewed:

- The [CHANGELOD.md file](https://github.com/demisto/content/blob/7d71a47813ab143689a5b0b77f98c6bcba01a29f/Integrations/Cisco-umbrella/CHANGELOG.md)
- The [Cisco-umbrella.yml file](https://github.com/demisto/content/blob/7d71a47813ab143689a5b0b77f98c6bcba01a29f/Integrations/Cisco-umbrella/Cisco-umbrella.yml)
Changes in yml file:
1. Command "domain" - The following context outputs were added (descriptions needs to be reviewed):
1.1 Domain.Admin.Country
1.2 Domain.Admin.Email
1.3 Domain.Admin.Name
1.4 Domain.Admin.Phone
1.5 Domain.Registrant.Country
1.6 Domain.Registrant.Email
1.7 Domain.Registrant.Name
1.8 Domain.Registrant.Phone
1.9 Domain.CreationDate
1.10 Domain.Registrar.Name
1.11 Domain.DomainStatus
1.12 Domain.UpdatedDate
1.13 Domain.ExpirationDate
2. Command "umbrella-get-whois-for-domain" - The following context outputs were added (descriptions needs to be reviewed):
Same context outputs mentioned above.

Guy F

**Comment by ygerowitz** [comment]
@guyfreund The attached Cisco-umbrella.yml file does not have **1.10 Domain.Registrar.Name**.

**Comment by ygerowitz** [comment]
@guyfreund It also seems that the Cisco-umbrella.yml file has not been reviewed. I've reviewed the list of the added outputs only, and a separate PR should opened for reviewing the entire thing.

**Comment by guyfreund** [comment]
> @guyfreund The attached Cisco-umbrella.yml file does not have **1.10 Domain.Registrar.Name**.

It has been updated, please look at the updated yml.

**Comment by ygerowitz** [comment]
@guyfreund Reviewed new context outputs. Added Domain.Registrar.Name (was missing).

**Comment by ygerowitz** [comment]
@guyfreund Just updated the changelog, please check for technical accuracy.

**Pull request closed without merging**